src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -1233,6 +1233,15 @@ impl ToBatch for LogicalAgg {
         let new_logical = self.clone_with_input(new_input);
         if self.group_key().is_empty() {
             Ok(BatchSimpleAgg::new(new_logical).into())
+        } else if self.group_key().iter().all(|group_by_idx| {
```

Comment:
I have check the code in backend https://github.com/singularity-data/risingwave/blob/main/src/batch/src/executor/sort_agg.rs, we should check if all the group key's datatype is Int32 or we can not use `SortAgg`.

Reply:
Alright, I used the input schema to check that